pages/index.js

```diff
@@ -49,32 +86,73 @@ export default function ShoppingItemsList({
             mode={mode}
           />
         )}
-
-        {mode !== "add" && (
-          <AddItemContainer>
-            <>
-              <p>...need anything else?</p>
+        {mode !== "add" ||
+          (!mode && (
+            <AddItemContainer>
+              <>
+                <p>...need anything else?</p>
+              </>
               <AddButton onClick={() => handleChangeMode("add")}>+</AddButton>
-            </>
-          </AddItemContainer>
+            </AddItemContainer>
+          ))}
+        {unpurchasedItems.length !== 0 && (
+          <div>
+            <StyledH2>
+              {/* {purchasedItems.length === 0 ? "We need " :  */}
+              There&#39;s {/* } */}
+              {unpurchasedItems.length} thing
+              {unpurchasedItems.length !== 1 ? "s" : ""}{" "}
+              {purchasedItems.length !== 0 ? "left to get" : "on the list"}:
+            </StyledH2>
+            <StyledList>
+              {unpurchasedItems.map((shoppingItem) => {
+                return (
+                  <ShoppingItem
+                    key={shoppingItem.id}
+                    shoppingItem={shoppingItem}
+                    shoppingItemsWithCategoryColor={
+                      shoppingItemsWithCategoryColor
+                    }
+                    backgroundColor={backgroundColor}
+                    onDeleteItem={onDeleteItem}
+                    onEditItem={() => setEditedItem(shoppingItem)}
+                    onChangeMode={() => handleChangeMode("edit")}
```

Comment:
I recommend to change the mode as part of your `onEditItem` function. This way you do not need to pass the `onChangeMode` function to the `ShoppingItem` component.

```suggestion
                  onEditItem={() => {
                      handleChangeMode("edit");
                      setEditedItem(shoppingItem);
                    }}
```

Reply:
This looks like a mess to me because I also use onChangeMode to reset the form to its default, i.e. empty and folded up state;
if I include the mode change (using setMode) in the handleEditItem function, I also have to include it for the Cancel button in the form component.
Thus, in principle I could include the mode change in the handleEditItem function but then I would have to move the useState to the app.js, and I would still have to pass it down to the form component. So this change doesn't buy me anything.
So for now, I will retain the current structure for the mode change.

Reply:
I feel like, you have misunderstood my suggestion.

You should definitely keep the `handleChangeMode` function. And yes, you are passing it down to your form.

But now we are looking at your shopping item. Basically, when the user clicks on the edit-button, you want to change the `mode` to "edit" and save the selected `shoppingItem` to the `editedItem` state. Currently, these operations are done by two separate functions (`handleChangeMode` and the setter-function `setEditedItem`). You are passing both down to your `shoppingItem` component and when the user clicks on the edit-button both functions get called after each other. 

The principle stays the same, we have two separate functions to take care of these operations. But instead of passing both down as a prop, you can simply combine them on this level (right in your `index.js`) in an anonymous function and pass these two functions down as ready-made package called `onEditItem`. 

I hope this explanation made it more clear. Otherwise, please do not hesitate to contact me and I will explain it to you. :)

Reply:
Hey, in fact, I just realised - I'm not even using onEdit in the ShoppingItem component, as all I do here is open the edit/add form. 
I only need it in the form. when I save the changes.
I have now removed onEdit from the shoppingItem component.